add handlers for dynamic registration

christianp opened the pull request:
Resolves #70.

See https://github.com/christianp/pylti1.3-django-example/tree/dynamic-registration for an example.

I've tried to add support for dynamic registration in a way that matches the rest of the package. I haven't written any unit tests or run pylint yet - I thought I'd ask for your opinion on the implementation before working out how to do that.

hmoffatt committed:
Don't send custom parameters if empty in deep linking result

christianp committed:
add handlers for dynamic registration

See https://github.com/christianp/pylti1.3-django-example/tree/dynamic-registration for an example.

hmoffatt commented:
It looks like the core will require asgiref for this, but it's not in the dependencies.

Is there a more generic sync_to_async() than using a Django library?

christianp commented:
@hmoffatt whoops, you're right! It's easy enough to remove async, so I've just done that.

yunusdemir commented:
I would love it for this PR to be merged. Is there any chance it will ever be reviewed @dmitry-viskov? Is there anything I could do to help get this merged?

christianp committed:
Changes to appease pylint

As well as formatting changes, there are a few methods which pylint said had too many positional arguments, so I've changed them to keyword arguments.

Anything using those methods will have to make sure the keyword arguments are named.

christianp committed:
pylintrc: fix overgeneral-exceptions

Apparently pylint wants you to get BaseException and Exception from builtins

christianp committed:
flake8 and pylint ignore migrations

They're automatically generated, so I don't think we need to lint them.

christianp committed:
move to src layout

christianp committed:
new metadata
